sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -24,22 +25,35 @@ use crate::io::uart::Uart;
 use crate::transport::{Capabilities, Capability, Transport, TransportError};
 use crate::util::usb::UsbBackend;
 
+pub mod c2d2;
 pub mod gpio;
 pub mod i2c;
 pub mod spi;
 pub mod uart;
 
 /// Implementation of the Transport trait for HyperDebug based on the
 /// Nucleo-L552ZE-Q.
-pub struct Hyperdebug {
+pub struct Hyperdebug<T: Flavor> {
     spi_names: HashMap<String, u8>,
     spi_interface: BulkInterface,
     i2c_names: HashMap<String, u8>,
     i2c_interface: BulkInterface,
     uart_ttys: HashMap<String, PathBuf>,
     inner: Rc<Inner>,
+    phantom: PhantomData<T>,
 }
 
+/// Trait allowing slightly different treatment of USB devices that work almost like a
+/// HyperDebug.  E.g. C2D2 and Servo micro.
+pub trait Flavor {
+    fn gpio_pin(inner: &Rc<Inner>, pinname: &str) -> Result<Rc<dyn GpioPin>>;
+    fn default_usb_vid() -> u16;
+    fn default_usb_pid() -> u16;
```

Comment:
nit: would these be better named "get_usb_*()"?

Reply:
The VID/PID can still be overridden by command line flags, so what is returned from these methods are the default values to use for the particular flavor, in case none are given on the command line.
I have called them get_default_usb_*().